src/GreenDonut/src/Core/TaskCacheOwner.cs

```diff
@@ -37,6 +39,11 @@ public TaskCacheOwner(ObjectPool<TaskCache> pool)
         /// </summary>
         public ITaskCache Cache => _cache;
 
+        public ITaskCache CacheByKey(string key)
```

Comment:
This would be wrong this one TaskCacheOwner is a handle for a single cache. Essentially, this follows the `MemoryOwner` in dotnet.